File: solarsystem/src/main/java/fail/stderr/usb/common/OrbitFunctions.java
package fail.stderr.usb.common;

import org.orekit.frames.Frame;
import org.orekit.orbits.KeplerianOrbit;
import org.orekit.orbits.PositionAngleType;
import org.orekit.time.AbsoluteDate;

public class OrbitFunctions {

  public static KeplerianOrbit createKeplerianOrbit(
    KeplerianOrbitParameters params,
    Frame parentFrame,
    AbsoluteDate refDate,
    double bodyMU
  ) {
    return new KeplerianOrbit(
      params.semiMajorAxis(),
      params.eccentricity(),
      params.inclination(),
      params.pa(),
      params.raan(),
      params.anomaly(),
      PositionAngleType.MEAN,
      parentFrame,
      refDate,
      bodyMU
    );
  }

}

File: solarsystem/src/main/kotlin/fail/stderr/usb/common/OrbitFunctions.kt
package fail.stderr.usb.common

import fail.stderr.usb.data.system.KeplerianOrbitalParametersData
import org.orekit.frames.Frame
import org.orekit.orbits.KeplerianOrbit
import org.orekit.orbits.PositionAngleType
import org.orekit.time.AbsoluteDate

class OrbitFunctions {
  companion object {
    fun createKeplerianOrbit(
      params: KeplerianOrbitalParametersData,
      parentFrame: Frame?,
      refDate: AbsoluteDate?,
      parentMU: Double
    ): KeplerianOrbit {
      return KeplerianOrbit(
        params.semiMajorAxis,
        params.eccentricity,
        params.inclination,
        params.perigreeArgument,
        params.raan,
        params.anomaly,
        PositionAngleType.MEAN,
        parentFrame,
        refDate,
        parentMU
      )
    }

  }
}